cryptographic_estimators/MREstimator/MRAlgorithms/support_minors.py

```diff
@@ -0,0 +1,261 @@
+
+    @optimal_parameter
+    def lv(self):
+        """
+        Return the optimal `lv`, i.e. no. of entries to guess in the solution
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.lv()
+            4
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_COEFFICIENTS_TO_GUESS)
+
+    @optimal_parameter
+    def b(self):
+        """
+        Return the optimal `b`, i.e. the degree of the linear variables in the Macaulay matrix
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.b()
+            1
+        """
+        return self._get_optimal_parameter(MR_LINEAR_VARIABLES_DEGREE)
+
+    @optimal_parameter
+    def nprime(self):
+        """
+        Return the optimal `nprime`, i.e. the number of columns to be selected
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.nprime()
+            4
+        """
+        return self._get_optimal_parameter(MR_REDUCED_NUMBER_OF_COLUMNS)
+
+    @optimal_parameter
+    def variant(self):
+        """
+        Return the optimal `variant`
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.nprime()
+            'strassen'
+        """
+        return self._get_optimal_parameter(MR_VARIANT)
+
+
+    def _expected_dimension_of_support_minors_equations(self,q, m, n, K, r, b):
+        """
+        Return the expected number of linearly independent support minors equations
+
+        """
+        if q == 2:
+            temp = 0
+            for j in range(1, b + 1):
+                temp += sum([(-1) ** (i + 1) * binomial(n, r + i) * binomial(m + i - 1, i) * binomial(K, j - i) for i in
+                             range(1, j + 1)])
+        else:
+            temp = sum(
+                [(-1) ** (i + 1) * binomial(n, r + i) * binomial(m + i - 1, i) * binomial(K + b - i - 1, b - i) for i in
+                 range(1, b + 1)])
+        return temp
+
+
+    def _dimension(self,q, n, K, r, b):
```

Comment:
Without in-depth knowledge:
Is temp really the dimension or is it rather log(temp)? It just looks like this value might get pretty large and the general convention is passing the logarithms in those cases, unless there are good reasons not to